docs/reference/content/reference/observables.md

```diff
@@ -0,0 +1,126 @@
+
+## Observer
+
+An [`Observer`]({{< apiref "org.mongodb.scala.Observer" >}}) provides the mechanism for receiving push-based notifications from the
+`Observable`.  Demand for these events is signalled by its `Subscription`.
+
+On subscription to an `Observable[TResult]` the `Observer` will be passed the `Subscription` via the 
+`onSubscribe(subscription: Subscription)`. Demand for results is signaled via the `Subscription` and any results are passed to the 
+`onNext(result: TResult)` method.  If there is an error for any reason the `onError(e: Throwable)` will be 
+called and no more events passed to the `Observer`. Alternatively, when the `Observer` has consumed all the results from the `Observable` 
+the `onComplete()` method is called.
+
+
+## Back Pressure
+
+In the following example, the `Subscription` is used to control demand when iterating an `Observable`. The default `Observer` implementation
+automatically requests all the data. Below we override the `onSubscribe` method custom so we can manage the demand driven iteration of the 
+`Observable`:
+
+ ```scala
+collection.find().subscribe(new Observer[Document](){
+
+  var batchSize: Long = 10
+  var seen: Long = 0
+  var subscription: Option[Subscription] = None
+  
+  override def onSubscribe(subscription: Subscription): Unit = {
+    this.subscription = Some(subscription)
+    subscription.request(batchSize)
+  }
+  
+  override def onNext(result: Document): Unit = {
+    println(document.toJson())
+    seen += 1
+    if (seen == batchSize) {
+      seen = 0
+      subscription.get.request(batchSize)
+    }
+  }
+
+  override def onError(e: Throwable): Unit = println(s"Error: $e")
+
+  override def onComplete(): Unit = println("Completed")
+})
+```
+## Observable Helpers
+
+The `org.mongodb.scala` package provides improved interaction with the 
+[Java `Observable`]({{< coreapi "com/mongodb/async/client/Observable.html" >}}) class via the 
+[`ScalaObservable`]({{< apiref "org.mongodb.scala.ScalaObservable" >}}) implicit class. The extended functionality includes simple 
+subscription via anonymous functions:
+
+```scala
+// Subscribe with custom onNext:
+collection.find().subscribe((doc: Document) => println(doc.toJson()))
+
+// Subscribe with custom onNext and onError
+collection.find().subscribe((doc: Document) => println(doc.toJson()),
+                            (e: Throwable) => println(s"There was an error: $e"))
+
+// Subscribe with custom onNext, onError and onComplete
+collection.find().subscribe((doc: Document) => println(doc.toJson()),
+                            (e: Throwable) => println(s"There was an error: $e"),
+                            () => println("Completed!"))
+```
+
+The `ObservableImplicits` trait also provides Monadic operators to make chaining and work with `Observable` instances simpler:
```

Comment:
Did you consider showing an example of how to use at least one of these implicits?